Update README (#419)
NabuCasa/support: README.md

# Nabu Casa Support Content

> This repo provides the content for the [Nabu Casa Zendesk support site](https://support.nabucasa.com/hc/en-us/). Each push to main will deploy the content to Zendesk.

Reviewer: ```suggestion
This repo provides the content for the [Nabu Casa Zendesk support site](https://support.nabucasa.com/hc/en-us/). Each push to the `main` branch will deploy the content to Zendesk.
```

## How does it work?

> A deploy script runs via GitHub Actions and uploads the content via the Zendesk REST API. Each PR builds a preview of on a replica of the Zendesk theme, which is then deployed to a preview URL via Netlify. This allows for easy review of changes before they are merged into the main branch.

Reviewer: ```suggestion
A [deploy script](./deploy.js) runs via [GitHub Actions](./.github/workflows/zendesk-deploy.yml) and uploads the content via the Zendesk REST API. Each PR builds a preview of a replica of the Zendesk theme, which is then deployed to a preview URL via Netlify. This allows for easy review of changes before they are merged into the main branch.
```

    - `_section.md` - (required) Metadata that describes the section for Zendesk
    - `{article}.md` - The article that will be sent to Zendesk. It also contains metadata for the article

## Metadata

There are some required and optional fields for each of the metadata files. The required fields are used to reference the Zendesk object, while the optional fields are used to populate the content or metadata for the object.

### Category (`_category.md`)

```yaml
---
zendesk:
  category_id: 24638797677853 # Required. The ID of the category in Zendesk
  name: Home Assistant Green # Required. The name of the category in Zendesk
  position: 1 # Optional. Defaults to 0
  description: The affordable Home Assistant Green is the... # Optional. The description of the category in Zendesk
---
```

### Section (`_section.md`)

```yaml
---
zendesk:
  section_id: 24737097675549 # Required. The ID of the section in Zendesk

>   name: Getting Started # Required. The name of the section in Zendesk

Reviewer 2: ```suggestion
  name: Getting started # Required. The name of the section in Zendesk
```

  position: 1 # Optional. The position of the section in Zendesk in relation to other sections
  description: All the things to get you started... # Optional. The description of the section in Zendesk
---
```

### Article (`_<article-slug>.md`)

```yaml
---
zendesk:
  article_id: 24737667232413 # Required. The ID of the article in Zendesk
  name: Getting started with Home Assistant Green # Required. The name of the article in Zendesk
  position: 2 # Optional. The position of the article in Zendesk in relation to other articles
---
```

# Creating / deleting content

This repo does not create or delete content in Zendesk. It only updates the content of existing articles, sections, and categories. There are a few things to note:

## Creating content

> If you want to create content, you must first make the content structure in Zendesk. This includes creating the category, section, and article(s). You do not need to populate any content in these objects. Once created, you must obtain the ID for each object. The easiest way to do this is to "preview" the article/section/category in Zendesk. The ID will be in the URL. For example, if the URL is `https://support.nabucasa.com/hc/en-us/articles/24737667232413-Setting-up-the-device`, the ID is `24737667232413`. You can then use this ID in the metadata for the object.
> 

Reviewer 2: ```suggestion
Before creating new content, you must first create the content structure in Zendesk.
1. In Zendesk, creating the category, section, and article(s). You do not need to populate any content in these objects. 2. Once created, obtain the ID for each object. 
    - To do this, "preview" the article/section/category in Zendesk and copy the ID from the URL.
    - For example, if the URL is `https://support.nabucasa.com/hc/en-us/articles/24737667232413-Setting-up-the-device`, the ID is `24737667232413`. 
4. Paste this ID into the metadata header of that object.

```

> ## Deleting content

Reviewer 2: ```suggestion
## Deleting articles, sections, or categories
```

> This repo does not delete content in Zendesk. If you want to delete content, you must do this in Zendesk. Once deleted, remove the relevant files from this repo to avoid errors.

Reviewer 2: ```suggestion
This repo can delete content of an article in Zendesk, but not an entire article, section, or category. If you want to delete these, you must do this in Zendesk. Once deleted, remove the relevant files from this repo to avoid errors.
```

```markdown
Correct link:
[Setting up the device](/hc/en-us/articles/24737667232413)
Incorrect link (using slug after the article ID):
[Setting up the device](/hc/en-us/articles/24737667232413-Setting-up-the-device)
Incorrect link (missing starting slash in URL):
[Setting up the device](hc/en-us/articles/24737667232413)
Incorrect link (using URL domain):
[Setting up the device](https://support.nabucasa.com/hc/en-us/articles/24737667232413)
```

Notice how the slug of the article is not included in the link. (e.g. use `/hc/en-us/articles/24737667232413` instead of `/hc/en-us/articles/24737667232413-Setting-up-the-device`). This is because Zendesk uses the title of the object as the slug which is subject to change with future updates. Conveniently, going to the URL without the slug will redirect to the correct URL with the slug. This will allow for easy linking to other articles without worrying about the slug changing in the future.

# How to contribute

> - Clone this repo into a docker container volume
> - Create a `.env` file at the root of the project, copying the contents of `.env.example`, replacing the values with your own
> - Run `npm run start` to start the Zendesk theme compiler and the preview server on port 8080
> - Make changes to the content in the `src/` folder
> - Make sure to run `npm run build` to build the content before pushing
> - Raise a PR to the main branch
> - Once the PR is merged, the content will be deployed to Zendesk

Reviewer 2: ```suggestion
- Clone this repo into a docker container volume.
- Create a `.env` file at the root of the project, copying the contents of `.env.example`, replacing the values with your own.
- Run `npm run start` to start the Zendesk theme compiler and the preview server on port 8080.
- Make changes to the content in the `src/` folder.
- Make sure to run `npm run build` to build the content before pushing.
- Raise a PR to the main branch.
- Once the PR is merged, the content will be deployed to Zendesk.
```

  description: The affordable Home Assistant Green is the... # Optional. The description of the category in Zendesk
---
```

### Section (`_section.md`)

```yaml
---
zendesk:
  section_id: 24737097675549 # Required. The ID of the section in Zendesk
  name: Getting started # Required. The name of the section in Zendesk
  position: 1 # Optional. The position of the section in Zendesk in relation to other sections
  description: All the things to get you started... # Optional. The description of the section in Zendesk
---
```

### Article (`_<article-slug>.md`)

```yaml
---
zendesk:
  article_id: 24737667232413 # Required. The ID of the article in Zendesk
  name: Getting started with Home Assistant Green # Required. The name of the article in Zendesk
  position: 2 # Optional. The position of the article in Zendesk in relation to other articles
---
```

# Creating / deleting content

This repo does not create or delete content in Zendesk. It only updates the content of existing articles, sections, and categories. There are a few things to note:

## Creating content

Before creating new content, you must first create the content structure in Zendesk.

> 1. In Zendesk, creating the category, section, and article(s). You do not need to populate any content in these objects. 2. Once created, obtain the ID for each object. 
>     - To do this, "preview" the article/section/category in Zendesk and copy the ID from the URL.
>     - For example, if the URL is `https://support.nabucasa.com/hc/en-us/articles/24737667232413-Setting-up-the-device`, the ID is `24737667232413`. 
> 4. Paste this ID into the metadata header of that object.

Reviewer 2: ```suggestion
1. In Zendesk, creating the category, section, and article(s). You do not need to populate any content in these objects.

2. Once created, obtain the ID for each object. 
    - To do this, "preview" the article/section/category in Zendesk and copy the ID from the URL.
    - For example, if the URL is `https://support.nabucasa.com/hc/en-us/articles/24737667232413-Setting-up-the-device`, the ID is `24737667232413`. 
3. Paste this ID into the metadata header of that object.
```

```markdown
Correct link:
[Setting up the device](/hc/en-us/articles/24737667232413)
Incorrect link (using slug after the article ID):
[Setting up the device](/hc/en-us/articles/24737667232413-Setting-up-the-device)
Incorrect link (missing starting slash in URL):
[Setting up the device](hc/en-us/articles/24737667232413)
Incorrect link (using URL domain):
[Setting up the device](https://support.nabucasa.com/hc/en-us/articles/24737667232413)
```

Notice how the slug of the article is not included in the link. (e.g. use `/hc/en-us/articles/24737667232413` instead of `/hc/en-us/articles/24737667232413-Setting-up-the-device`). This is because Zendesk uses the title of the object as the slug which is subject to change with future updates. Conveniently, going to the URL without the slug will redirect to the correct URL with the slug. This will allow for easy linking to other articles without worrying about the slug changing in the future.

# How to contribute

- Clone this repo into a docker container volume.

> - Create a `.env` file at the root of the project, copying the contents of `.env.example`, replacing the values with your own.

Reviewer: No contributor should ever need this .env file, ideally the file itself would be removed as this is only used by the github action and should never be run manually.
